docs/rfc/active/000-ml_models/README.md

```diff
@@ -0,0 +1,121 @@
+- Start Date: 08/18/2020
+- RFC PR: https://github.com/linkedin/datahub/pull/1812
+- Implementation PR(s): https://github.com/linkedin/datahub/pull/1721
+
+# Machine Learning Models
+
+## Summary
+
+Adding support for trained machine learning models and features metadata cataloging and enabling search and discovery over them. This is a step towards organizing the essential facts of machine learning models in a structured way leading to responsible democratization of machine
+learning and related artificial intelligence technology. The work is inspired by Google's model card [paper](https://arxiv.org/pdf/1810.03993.pdf).
+
+## Motivation
+
+We need to model ML model metadata for transparent model reporting. Below are some of the reasons why storing machine learning model metadata is important:
+- Drawing boundaries around a model's capabilities and limitations: There is a need to store the conditions under which a model performs best and most consistently and if it has some blind spots. It helps potential users of the models be better informed on which models are best for their specific purposes. Also, it helps minimize usage of machine learning models in contexts for which they are not well suited.
+- Metrics and Limitations: A model’s performance can be measured in countless ways, but we need to catalog the metrics that are most relevant and useful. Similarly there is a need to store a model's potential limitations that are most useful to quantify.
+- Ensure comparability across models in a well-informed way: Modeling metadata of ML models allows us to compare candidate models' results across not only traditional evaluation metrics but also along the axes of ethical, inclusive, and fair
+considerations.
+- Promote reproducibility: Often a model is trained on transformed data, there are some preprocessing steps involved in transforming the data e.g. centering, scaling, dealing with missing values, etc. These transforms should be stored as part of model metadata to ensure reproducibility. 
+
+## Detailed design
+![high level design](high_level_design.png)
+
+
+As shown in the above diagram, machine learning models are using machine learning features as inputs. These machine learning features
+could be shared across different machine learning models. In the example sketched above, `ML_Feature_1` and `ML_Feature_2` are used as inputs for `ML_Model_A` while `ML_Feature_2`, `ML_Feature_3` and `ML_Feature_4` are inputs for `ML_Model_B`.
+
+### URN Representation
+We'll define two [URNs](../../../what/urn.md): `MLModelUrn` and `MLFeatureUrn`.
+These URNs should allow for unique identification of machine learning models and features, respectively. Machine learning models, like datasets, will be identified by combination of standardized platform urn, name of the model and the fabric type where the model belongs to or where it was generated. 
```

Comment:
Was asked in the models PR as well. Here is the reply given by @arunvasudevan - "In the same way how we have DataPlatform for the Dataset like DataLake, StreamPlatform, etc.... This could be some of the DataPlatform for the MlModel in our case we use it like DataSciencePlatform it could also be TensorFlow as well...."

@arunvasudevan can share more insights here

Reply:
Sorry for my late response here....
This is added to identify different ML Platforms within the Org...In our case definitely we have a few ingrown ML platforms which different sections of the org has adopted. Making it part of the Urn would also help us to move MLModels from One platform (Currently In house) to a different platform (say Tensorflow) in future with the same Model name.

Reply:
Updated the doc to reflect the same.

Reply:
Also representing the platform as an urn will enable us to associate additional metadata to it.